### PR TITLE
Resolved IDENTITY-4393

### PR DESCRIPTION
**Pull request by IndunilRathnayake**

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/AuthenticatedUser.java [hunk 1]
```diff
@@ -90,15 +91,15 @@ public static AuthenticatedUser createLocalAuthenticatedUserFromSubjectIdentifie
...
-                authenticatedUser.setUserStoreDomain(subjectIdentifierSplits[0]);
-                authenticatedUser.setUserName(MultitenantUtils.getTenantAwareUsername(subjectIdentifierSplits[1]));
-            } else {
-                authenticatedUser.setUserName(MultitenantUtils.getTenantAwareUsername(authenticatedSubjectIdentifier));
-            }
+            String[] subjectIdentifierSplits =
+                    authenticatedSubjectIdentifier.split(CarbonConstants.DOMAIN_SEPARATOR, 2);
+            authenticatedUser.setUserStoreDomain(subjectIdentifierSplits[0]);
+            authenticatedUser.setUserName(MultitenantUtils.getTenantAwareUsername(subjectIdentifierSplits[1]));
+        } else if (UserCoreUtil.getDomainFromThreadLocal() != null && !UserCoreUtil.getDomainFromThreadLocal().isEmpty()) {
```
Comment (Pushpalanka) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/AuthenticatedUser.java]: Can use StringUtils.isNotEmpty()

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/AuthenticatedUser.java [hunk 2]
```diff
@@ -90,15 +91,15 @@ public static AuthenticatedUser createLocalAuthenticatedUserFromSubjectIdentifie
...
+            authenticatedUser.setUserStoreDomain(UserCoreUtil.getDomainFromThreadLocal());
+            authenticatedUser.setUserName(MultitenantUtils.getTenantAwareUsername(authenticatedSubjectIdentifier));
         } else {
+            authenticatedUser.setUserStoreDomain(UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME);
```
Comment (Pushpalanka) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/AuthenticatedUser.java]: Don't set the default primary domain name as this domain name can be configured as well. There is a Util method at IdentityUtil class to get the primary domain name. Use it.

**Comment by cdwijayarathna** [comment]
You need to send PRs for identity-framework repo or related extension repo, we are no longer using this carbon-identity repo

**Comment by IndunilRathnayake** [comment]
Moved to identity-framework

**Pull request closed without merging**